src/guide/project-maintenance.md

```diff
@@ -0,0 +1,66 @@
+
+~~~
+cd project-merge
+git pull upstream master
+git push
+git switch -c Merge_mainline
+git fetch mainline
+git merge mainline/master
+~~~
+
+We start by updating the local fork with the latest changes from the main project repository. Note that we then create a new branch in which the merge will happen.
+
+The commands above will likely run without a hitch up until the `git merge`. This is where you need to combine the changes that were made in mainline with the changes that have been made in your project repository. If there are no conflicts you're in luck, then the merge will be completely automated and you will end up with a commit ready to push. If there are conflicts however you'll need to manually go through the files where the conflicts are and make sure you select the correct version for each change. Using `git status` you can see what files that need to be merged. Depending on how much code your project has touched this can be quite a bit of work.
+
+Once you have a working version of your merged code (make sure you test it properly) you're ready to create the commit and push. Please note that `git commit` is only needed if there were conflicts. If the changes were successfully merged by `git merge`, you're already ready to push.
+
+~~~
+git commit -m "Merge"
+git push --set-upstream origin Merge_mainline
+~~~
+
+Now it's time to create the PR on GitHub. Just opening the PR page in your browser will most often be enough to see a message about new code that was pushed to your personal fork, with a question if you want to create a PR. You do.
+
+Make sure the name of the PR starts with "Merge". You may have noticed that when you integrate a "normal" PR into an OpenJDK repository, all commits that have been done in that PR will be squashed into a single commit. For normal changes this is a good thing as each PR normally corresponds to a single JBS issue, but for a merge it would be highly undesirable to squash all the different commits that you pull in from mainline. A PR with a name that starts with "Merge" won't be squashed. That means that all the changes that you brought over will remain separate changes.
+
+Wether a merge requires a review or not is up to your project lead to decide. Many projects don't require this so the GitHub bots will allow you to push the merge as soon as the [GHA](#github-actions)s are done. (They actually allow you to push even before the GHAs are done, but that's in general not a good idea.)
+
+Once the PR has been integrated, you can clean up your fork and its clone in preparation for the next merge.
+
+~~~
+git switch master
+git branch -d Merge_mainline
+git push -d origin Merge_mainline
+~~~
+
+These commands will remove the temporary branch that we created to perform the merge.
```

Comment:
The github UI has a button to delete the branch after you have integrated the PR.

Reply:
Added a sentence about this.